srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/SignedResult.java

```diff
@@ -18,96 +27,257 @@
+        setPatientDfn(patientDfn);
+        setCptCode(cptCode);
+        // See getSignatureTimestamp() for why we enforce 0 millis of second.
+        setSignatureTimestamp(signatureTimestamp.withMillisOfSecond(0).toDate());
+        setInputs(ImmutableMap.copyOf(inputs));
+        setOutcomes(ImmutableMap.copyOf(outcomes));
     }
     
     /**
+     * The object's surrogate primary key. Don't show this to the user.
+     */
+    @Id
+    public int getId()
+    {
+        return fId;
+    }
+
+    /**
+     * For reflection-based construction only. Business code should never modify
+     * the surrogate key as it is generated from the database.
+     */
+    void setId(final int id)
+    {
+        fId = id;
+    }
+    
+    /**
+     * Returns the associated historical calculation information.
+     * @return never null
+     */
+    @OneToOne(fetch = FetchType.EAGER, optional = false)
+    @JoinColumn(name = "run_id")
+    @MapsId
+    public HistoricalCalculation getHistoricalCalculation()
+    {
+        return fHistoricalCalculation;
+    }
+    
+    /**
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
+     */
+    void setHistoricalCalculation(final HistoricalCalculation historicalCalc)
+    {
+        fHistoricalCalculation = Objects.requireNonNull(historicalCalc);
+    }
+
+    /**
      * Returns the DFN of the associated patient.
      */
+    @Basic
     public int getPatientDfn()
     {
         return fPatientDfn;
     }
-
+    
     /**
-     * Returns the name of the associated surgical specialty.
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
      */
-    public String getSpecialtyName()
+    void setPatientDfn(final int patientDfn)
     {
-        return fSpecialtyName;
+        fPatientDfn = patientDfn;
     }
     
     /**
      * Returns the CPT code of the associated procedure, if there was one.
      * @return an Optional containing the CPT code, if applicable
      */
+    @Transient  // Hibernate uses getCptCodeNullabe() instead
     public Optional<String> getCptCode()
     {
         return fCptCode;
     }
     
     /**
-     * The DateTime at which the user started the calculation. We do not track
-     * milliseconds, so the millisOfSecond field will always be 0 to facilitate
-     * comparisons.
+     * Sets the cptCode, checking preconditions.
+     * @throws IllegalArgumentException if the cptCode is not {@link
+     * Procedure#CPT_CODE_LENGTH} characters
+     */
+    private void setCptCode(final Optional<String> optional)
+    {
+        if (optional.isPresent())
+        {
+            Preconditions.requireWithin(
+                    optional.get(), Procedure.CPT_CODE_LENGTH, Procedure.CPT_CODE_LENGTH);
+        }
+
+        fCptCode = optional;
+    }
+    
+    /**
+     * As {@link #getCptCode()}, but represents a missing CPT Code as null. Purely to
```

Comment:
Same as the one comment before, is there supposed to be more information with the "As"?